libs/build_met.sh

```diff
@@ -0,0 +1,136 @@
+    module load g2c
+    module load zlib
+    module load jasper
+    module load png
+    module load gsl
+    module list
+    set -x
+    prefix="${PREFIX:-"/opt/modules"}/$compiler/$mpi/$name/$version"
+    if [[ -d $prefix ]]; then
+	[[ $OVERWRITE =~ [yYtT] ]] && ( echo "WARNING: $prefix EXISTS: OVERWRITING!";$SUDO rm -rf $prefix ) \
+	    || ( echo "WARNING: $prefix EXISTS, SKIPPING"; exit 1 )
+    fi
+else
+
+    prefix=${MET_ROOT:-"/usr/local"}
+
+fi
+
+export MET_BASE=$prefix/share/met
+
+if [[ ! -z $mpi ]]; then
+    export FC=$MPI_FC
+    export CC=$MPI_CC
+    export CXX=$MPI_CXX
+else
+    export FC=$SERIAL_FC
+    export CC=$SERIAL_CC
+    export CXX=$SERIAL_CXX
+fi
+
+export F77=$FC
+export FFLAGS="${STACK_FFLAGS:-} ${STACK_met_FFLAGS:-}"
+export CFLAGS="${STACK_CFLAGS:-} ${STACK_met_CFLAGS:-}"
+
+export MET_NETCDF=${NetCDF_ROOT}
+export MET_HDF5=${HDF5_ROOT}
+export MET_BUFRLIB=${bufr_ROOT}/lib
+export MET_GRIB2CLIB=${g2c_ROOT}/lib
+export MET_GRIB2CINC=${g2c_ROOT}/include
+export MET_GSL=${GSL_ROOT}
+export BUFRLIB_NAME=-lbufr_4
+export GRIB2CLIB_NAME=-lg2c
+export LIB_JASPER=${JASPER_ROOT}/lib64
+export LIB_LIBPNG=${PNG_ROOT}/lib64
+export LIB_Z=${ZLIB_ROOT}/lib
+
+LDFLAGS1="-Wl,--disable-new-dtags"
+LDFLAGS2="-Wl,-rpath,${MET_NETCDF}/lib:${MET_HDF5}/lib:${MET_BUFRLIB}"
+LDFLAGS3="-Wl,-rpath,${MET_GRIB2CLIB}:${MET_PYTHON}/lib:${MET_GSL}/lib"
+LDFLAGS4="-L${LIB_JASPER} -L${MET_HDF5}/lib -L${LIB_LIBPNG} -L${LIB_Z}"
+LDFLAGS5="-L${I_MPI_ROOT}/lib64"
+if [[ -z $mpi ]]; then
+  export LDFLAGS="${LDFLAGS1:-} ${LDFLAGS2:-} ${LDFLAGS3:-} ${LDFLAGS4:-}"
+else
+  export LDFLAGS="${LDFLAGS1:-} ${LDFLAGS2:-} ${LDFLAGS3:-} ${LDFLAGS4:-} ${LDFLAGS5:-}"
+fi
+  export LIBS="-lhdf5_hl -lhdf5 -lz"
+
+
+export CFLAGS+="-D__64BIT__"
+export CXXFLAGS+="-D__64BIT__"
+
+cd ${HPC_STACK_ROOT}/${PKGDIR:-"pkg"}/${pkg_name}
+curr_dir=$(pwd)
+echo "$(pwd)"
+
+echo "MET Configuration settings..."
+printenv | egrep "^MET_" | sed -r 's/^/export /g'
+echo "LDFLAGS = ${LDFLAGS}"
+
+echo "./configure --prefix=$prefix BUFRLIB_NAME=${BUFRLIB_NAME} GRIB2CLIB_NAME=${GRIB2CLIB_NAME} --enable-grib2 --enable-python"
+./configure --prefix=$prefix BUFRLIB_NAME=${BUFRLIB_NAME} GRIB2CLIB_NAME=${GRIB2CLIB_NAME} --enable-grib2 --enable-python
+
+ret=$?
+if [ $ret != 0 ]; then
+    echo "configure returned with non-zero ($ret) status"
+    exit 1
+fi
+
+echo "make > make.log 2>&1"
+make > make.log 2>&1
+ret=$?
+if [ $ret != 0 ]; then
+    echo "make returned with non-zero ($ret) status"
+    exit 1
+fi
+
+echo "make install > make_install.log 2>&1"
+make install > make_install.log 2>&1
+ret=$?
+if [ $? != 0 ]; then
+    echo "make install returned with non-zero ($ret) status"
+    exit 1
+fi
+
+echo "make test > make_test.log 2>&1"
+make test > make_test.log 2>&1
+ret=$?
+if [ $? != 0 ]; then
+    echo "make test returned with non-zero ($ret) status"
+    exit 1
+fi
+
+export PATH=${HPC_STACK_ROOT}/${PKGDIR:-"pkg"}/${pkg_name}:${PATH}
+echo "Finished compiling at `date`"
```

Comment:
Need to add a module for MET.

Reply:
@aerorahul 
> Need to add a module for MET.

Can you please explain further?  I think I don't fully understand.

As mentioned in the text for this pull request:
> I created lua files for MET and METplus, however, I wasn't sure if that was necessary or not. I am not very familiar with lua files and do not know how to test them, so they have not yet been tested.

These files are at:
modulefiles/mpi/compilerName/compilerVersion/mpiName/mpiVersion/met/met.lua 
modulefiles/compiler/compilerName/compilerVersion/metplus/metplus.lua

If this is not what you are referring to, could you please provide some additional guidance?